rero_ils/modules/documents/views.py

```diff
@@ -113,46 +111,76 @@ def cover(isbn):
+    """Import record from BNFr given a isbn 13 without dashes.
+
+    See: https://catalogue.bnf.fr/api/test.do
+    """
     bnf_url = current_app.config['RERO_ILS_APP_IMPORT_BNF_EAN']
     try:
-        with urlopen(bnf_url % ean) as response:
-            if response.status != 200:
-                abort(502)
-            # read the xml date from the HTTP response
-            xml_data = response.read()
-
-            # create a xml file in memory
-            xml_file = six.BytesIO()
-            xml_file.write(xml_data)
-            xml_file.seek(0)
-
-            # get the record in xml if exists
-            # note: the request should returns one record max
-            xml_record = next(split_stream(xml_file))
-
-            # convert xml in marc json
-            json_data = create_record(xml_record)
-
-            # convert marc json to local json format
-            record = unimarctojson.do(json_data)
-            response = {
-                'metadata': record
-            }
-            return jsonify(response)
+        with requests.get(bnf_url.format(ean)) as response:
+            if not response.ok:
+                status_code = 502
+                response = {
+                    'metadata': {},
+                    'errors': {
+                        'code': status_code,
+                        'title': 'The BNF server returns a bad status code.',
+                        'detail': 'Status code: {}'.format(
+                            response.status_code)
+                    }
+                }
+                current_app.logger.error(
+                    '{}: {}'.format(
+                        response.get('title'), response.get('detail')))
+
+            else:
+                # read the xml date from the HTTP response
+                xml_data = response.content
+
+                # create a xml file in memory
+                xml_file = six.BytesIO()
+                xml_file.write(xml_data)
+                xml_file.seek(0)
+
+                # get the record in xml if exists
+                # note: the request should returns one record max
+                xml_record = next(split_stream(xml_file))
+
+                # convert xml in marc json
+                json_data = create_record(xml_record)
+
+                # convert marc json to local json format
+                record = unimarctojson.do(json_data)
+                response = {
+                    'metadata': record
+                }
+                status_code = 200
     # no record found!
     except StopIteration:
+        status_code = 404
         response = {
-            'record': {}
+            'metadata': {},
+            'errors': {
+                'code': status_code,
+                'title': 'The EAN was not found on the BNF server.'
+            }
         }
-        return jsonify(response), 404
     # other errors
-    except Exception:
-        sys.stdout.flush()
+    except Exception as e:
```

Comment:
Recommendation from python: Use more verbose names for your variables for clarity. `e`--> `err` ?